extensions/SpyroStudios/Arrays.js

```diff
@@ -0,0 +1,497 @@
+            text: "column count of 3D array [NAME]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          },
+          {
+            opcode: "arrayDepthCount",
+            text: "depth count of 3D array [NAME]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          },
+          {
+            opcode: "duplicateArray3D",
+            text: "duplicate 3D array [NAME] called [NEWNAME]",
+            blockType: BlockType.COMMAND,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              },
+              NEWNAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "newArray"
+              }
+            }
+          },
+          {
+            opcode: "countOccurrencesOfItem3D",
+            text: "count # of item [ITEM] in 3D array [NAME]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              ITEM: {
+                type: ArgumentType.STRING,
+                defaultValue: "item"
+              },
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          }
+        ]
+      };
+    }
+
+    // Implement methods for 2D and 3D arrays
+
+    // 2D Array Methods
+    create2DArray(args, util) {
+      const name = `${args.NAME}_2D`;
+      const rows = Math.max(1, args.ROWS);
+      const columns = Math.max(1, args.COLUMNS);
+      const array = [];
+      for (let i = 0; i < rows; i++) {
+        array[i] = new Array(columns).fill(0);
+      }
+      util.target.variables[name] = array;
+      return array;
+    }
+
+    set2DArrayValue(args, util) {
+      const row = Math.max(0, args.ROW - 1);
+      const column = Math.max(0, args.COLUMN - 1);
+      const value = args.VALUE;
+      const name = `${args.NAME}_2D`;
+      const array = util.target.variables[name];
+      if (array && array.length > row && array[row].length > column) {
+        array[row][column] = value;
+      }
+    }
+
+    get2DArrayValue(args, util) {
+      const row = Math.max(0, args.ROW - 1);
+      const column = Math.max(0, args.COLUMN - 1);
+      const name = `${args.NAME}_2D`;
+      const array = util.target.variables[name];
+      if (array && array.length > row && array[row].length > column) {
+        return array[row][column];
+      }
+      return 0;
+    }
+
+    arrayExists2D(args, util) {
+      const name = `${args.NAME}_2D`;
+      return !!util.target.variables[name];
+    }
+
+    arrayRowCount2D(args, util) {
+      const name = `${args.NAME}_2D`;
+      if (!!util.target.variables[name]) {
```

Comment:
ok this is my fault; in if statements you don't need the !! because it's already checking a condition